BraintreeCore/src/main/java/com/braintreepayments/api/APIClient.java

```diff
@@ -7,18 +7,18 @@
 
 import java.lang.ref.WeakReference;
 
-class TokenizationClient {
+class APIClient {
```

Comment:
Why are we opting to rename this?

Reply:
Mostly to better align with iOS. The discussion started [here](https://github.com/braintree/braintree_android/pull/364#discussion_r641822294) on another PR.

Reply:
We did talk about renaming tokenizeRest and tokenizeGraphQL to `vault()` as well if that would make more sense. I forgot to add that change.

Reply:
There is also a really peculiar arch issue with tokenizeGraphQL and tokenizeREST. This class used to have a circular dependency on the `Card` module. To eliminate, we made `tokenizeGraphQL` take a `JSONObject` payload, and `tokenizeREST` still takes a `PaymentMethod` to get the correct `POST` route for a particular payment method.

It might make sense to go with `vaultUsingGraphQL` and `vaultUsingREST` here, but there's still something to figure out internally before we can make it a single method. I believe Susan was saying that tokenization doesn't always happen.

Reply:
The vault naming still seems a little confusing to me. For the Venmo case it makes sense because we only call `tokenizeREST` for vaulting, but it seems like we call `tokenizeREST` in `PayPalClient#onBrowserSwitchResult` for all PayPal browser switch results. 

It looks like iOS has separate [tokenize](https://github.com/braintree/braintree_ios/blob/68ca8ee70daa279ebfd5bbea62e52212f948d6d7/Sources/BraintreeVenmo/BTVenmoDriver.m#L160) and [vault](https://github.com/braintree/braintree_ios/blob/68ca8ee70daa279ebfd5bbea62e52212f948d6d7/Sources/BraintreeVenmo/BTVenmoDriver.m#L208) methods in the payment method drivers that call the underlying `BTAPIClient` `POST`  methods. That seems to make sense, but I'm not sure it applies the same way here unless we move the `POST` and `GET` methods from the `HTTPClient` to `APIClient` and call them in a similar way from the payment method clients.

Reply:
Yeah the naming confuses me a bit also. I think we could legitimately defer the `vault()` naming change since `APIClient` in iOS doesn't abstract HTTP, and this `APIClient` keeps the HTTP communication under the hood. There isn't a direct 1-to-1 correlation between the two classes, but we could one day make it a point to align APIClient for both iOS and Android (and maybe JS 🤷‍♂️ ) since they're internal classes.